vowpalwabbit/global_data.cc

```diff
@@ -2,18 +2,24 @@
 // individual contributors. All rights reserved. Released under a BSD (revised)
 // license as described in the file LICENSE.
 
-#include <stdio.h>
-#include <float.h>
-#include <errno.h>
+#include <cstdio>
+#include <cfloat>
+#include <cerrno>
 #include <iostream>
 #include <sstream>
-#include <math.h>
-#include <assert.h>
+#include <cmath>
+#include <cassert>
 
 #include "global_data.h"
 #include "gd.h"
 #include "vw_exception.h"
 
+#ifdef _WIN32
```

Comment:
does there have to be an else to this?

Reply:
It seems the socket symbols are coming in from somewhere else on linux, I'll include them for safety though